torch/csrc/jit/script/compiler.cpp

```diff
@@ -34,23 +34,127 @@ using ListAttributeMap = std::unordered_map<
     std::string,
     std::pair<const std::vector<double>, std::string>>;
 
+// Keep track of environment as we descend down nested control
+// structures.
+struct Environment {
+  Environment(Block* b, std::shared_ptr<Environment> next = nullptr)
+      : b(b), next(next) {}
+
+  std::vector<std::string> captured_inputs;
```

Comment:
Hmm, we don't have a type for frontend identifiers? These are not symbols right?

Reply:
No, they are plain strings/Ident trees that just hold on to the strings